sktime/utils/validation/__init__.py

```diff
@@ -38,12 +44,44 @@ def check_n_jobs(n_jobs):
         return n_jobs
 
 
-def check_window_length(window_length, name="window_length"):
+def check_window_length(window_length, n_timepoints=None, name="window_length"):
     """Validate window length"""
+    """
+    Parameters
+    ----------
+    window_length: positive int or positive float
+        The number of training set used for splitting
+        Window length for transformed feature variables
+
+    Returns
+    -------
+    window_length: int
+    """
     if window_length is not None:
-        if not is_int(window_length) or window_length < 1:
+
```

Comment:
Looking at this function, it still seems a bit complicated to me. I would try to simplify it further. One thing I would try to get rid of is the `valid` flag.

Reply:
I have refactored the function and removed `valid` from it